### PR TITLE
Rule: Domain Trust Discovery and new IP checking domain

### PR DESCRIPTION
**Pull request by johnk3r**
#174

### CODE REVIEW

File: collection/network/domain-trust.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+rule:
+  meta:
+    name: Detect Domain Trust Discovery
```
Comment (mr-tz) [inline collection/network/domain-trust.yml:3]: Ideally this rules completes the sentence "this function may...". How about `get domain trust relationships`?

File: collection/network/domain-trust.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    scope: function
+    att&ck:
+      - Discovery::Domain Trust Discovery  [T1482]
+    examples:
+      - 0796f1c1ea0a142fc1eb7109a44c86cb:0x40222F
+      - 0731679c5f99e8ee65d8b29a3cabfc6b:0x40408E
+  features:
+    - or:
+      - api: DsEnumerateDomainTrusts
+      - string: /nltest.exe/i
```
Comment (mr-tz) [inline collection/network/domain-trust.yml:15]: How about removing the `.exe` to be a little more flexible?
Also should we add `DOMAIN_TRUSTS` as `nltest` can be used for various tasks?

File: collection/network/capture-public-ip.yml [hunk 3]
```diff
@@ -23,3 +23,4 @@ rule:
           - string: /checkip\.amazonaws\.com/
           - string: /icanhazip\.com/
           - string: /wtfismyip\.com/text/
+          - string: /api\.myip\.com/
```
Comment (mr-tz) [inline collection/network/capture-public-ip.yml:26]: 👍

**Review by mr-tz — CHANGES_REQUESTED** [review]
Great, thank you. I only have a few comments.

**Comment by johnk3r** [comment]
Hello @mr-tz ,

tks for the suggestions. I changed the logic of the rule to improve detection and avoid false positives.

The only thing is to note that leaving only "nltest" detection did not occur. So I followed some examples of rules that use the character "*".

Also, if the malware contains two spaces (or more)  between commands the detection does not occur. So I left the strings separately. Do you have any smarter idea to avoid this "bypass"?

**Comment by mr-tz** [comment]
That looks great. I tried the below modified rule on the test samples:
- combines `nltest` and domain arguments under an `and` statement
- removed trailing `.*` from `nltest` (capa assumes leading and trailing wildcards)
- escaped `/` character

```yaml
  features:
    - or:
      - and:
        - string: /nltest/i
        - or:
          - string: /\/domain_trusts/i
          - string: /\/dclist/i
          - string: /\/all_trusts/i
      - api: DsEnumerateDomainTrusts
```

What do you think of these changes?

**Comment by johnk3r** [comment]
Great.

I adjusted the rule again.

**Review by mr-tz — APPROVED** [review]
lgtm! thank you!

**Pull request merged**